### ISSUE TITLE
Need help with the rewrite

### ISSUE THREAD

**Issue by jpillora** [jpillora/xhook#119]
Hey all

A few months ago, I started work on modernizing xhook. I wrote this branch https://github.com/jpillora/xhook/tree/es6-2020-refactor which uses ES6 and webpack, and the code is much cleaner. It contains a few fixes for some long standing bugs.

I'm looking for someone to either:

* Fix the testcafe test suite OR
* Write a new test suite  

I've signed up for an open-source profile on BrowserStack for xhook, and I can provide the API key to anyone who open's a PR with the tests passing **locally**.

Next step is to setup Github actions to run on every commit, once we have a test suite running and we're confident things work in most browsers. We can release this new version, along with its fixes

**Comment by TrickyPi** [comment]
Hi, I'm using `xhook` in some projects and it's really useful. I would like to solve the remaining tasks, do we have a new task list?

**Comment by jpillora** [comment]
Hi @TrickyPi, thanks for the offer to help! Please checkout branch https://github.com/jpillora/xhook/tree/es6-2020-refactor 

Run `npm install` and `npm run build` the output `xhook.js` file should be working

But before releasing this version, i'd like to have automated tests working. I'd like to have a test suite run against multiple **real** browsers. https://testcafe.io/ does this, browserling is another, cypress is probably the most modern one (example on GHA https://docs.cypress.io/guides/continuous-integration/github-actions)

The main task is to setup a passing test suite which can be executed by github actions to confirm changes don't break existing functionality

Once this is done, it should be fairly straight forward to start fixing the other issues that have accumulated over the years 😅

**Comment by jpillora** [comment]
Before embarking on this, please post your plan here and ill double check it'll work. Also, once we merge your PR, ill make you a contributor :)

**Comment by TrickyPi** [comment]
> Run `npm install` and `npm run build` the output `xhook.js` file should be working

yeah, it works.

>  I'd like to have a test suite run against multiple real browsers. https://testcafe.io/ does this, browserling is another, cypress is probably the most modern one (example on GHA https://docs.cypress.io/guides/continuous-integration/github-actions)

Could we adopt `playwright`(https://github.com/microsoft/playwright) as our e2e framework?

**Comment by TrickyPi** [comment]
Here is my simple plan.

1. migrate test to playwright
2. setup github action
3. add `type` file for  `xhook` (maybe we can refactor code via typescript)

**Comment by jpillora** [comment]
Play write sounds good! I’ll be watching the PRs :)

**Comment by jpillora** [comment]
Oh yea, if you don’t mind, please switch to typescript too 👌

**Comment by TrickyPi** [comment]
Ok
